src/jit/lower.cpp

```diff
@@ -2427,8 +2433,13 @@ void Lowering::LowerCompare(GenTree* cmp)
                 // situations. In particular this include CALL, sometimes the JIT unnecessarilly widens
                 // the result of bool returning calls.
                 //
+                bool removeCast =
+#ifdef _TARGET_ARM64_
+                    (op2Value == 0) &&
```

Comment:
This should probably also have ` && cmp->OperIs(GT_EQ, GT_NE)`

It could be changed to `((op2Value == 0) || isPow2(op2Value))  && cmp->OperIs(GT_EQ, GT_NE)` to support `==/!= true` forms, with corresponding changes below.